### DISCUSSION TITLE
Query data from Mimir (using python or any other mode)

### DISCUSSION THREAD

**Discussion by velu4689** [grafana/mimir#5612] [General]
Hi Team,

Currently am using Mimir + Grafana to visualize the data. Is there any additional tool (python/curl) that can be used for querying the data out of Mimir ?

Thanks

**Comment by pstibrany** [comment]
Hi,

yes, any tool that can access HTTP endpoints can query the data.

```bash
#!/bin/sh

URL=http://mimir-url/api/prom
TENANT=anonymous

curl -s -H "X-Scope-OrgID: $TENANT" "$URL/api/v1/query" -d query="$1" | jq .
```

For list of endpoints, please see https://grafana.com/docs/mimir/latest/references/http-api/. For querying see "Instant Query" and "Instant Range Query".

**Comment by velu4689** [comment]
@pstibrany : Thanks. But we're observing a weird behavior where, the number of samples returned increases incrementally every 5 mins & then again drops to zero.

Also, how do we add the scrape time interval inside "--selector" while using Mimirtool ?

**Reply by pstibrany** [reply]
> But we're observing a weird behavior where, the number of samples returned increases incrementally every 5 mins & then again drops to zero.

Hi, to better understand your problem can you please provide an example of result that you’re seeing, and exact request (url, parameters) that you use to query Mimir?
